src/components/container/RelativeActivity/RelativeActivity.tsx

```diff
@@ -0,0 +1,184 @@
+import { faChevronDown, faChevronUp } from "@fortawesome/free-solid-svg-icons";
+import { add, startOfDay } from "date-fns";
+import React, { useContext, useEffect } from "react";
+import { useState } from "react";
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+import getDayKey from "../../../helpers/get-day-key";
+import { DailyDataQueryResult, queryDailyData } from "../../../helpers/query-daily-data";
+import { ActivityMeter, CardTitle, DateRangeContext, } from "../../presentational";
+import "./RelativeActivity.css"
+import language from "../../../helpers/language";
+import { useInitializeView } from "../../../helpers/Initialization";
+import { ColorDefinition } from "../../../helpers/colors";
+
+export interface RelativeActivityProps {
+    dataTypes: RelativeActivityDataType[];
+    previewState?: "Default";
+    title?: string;
+    innerRef?: React.Ref<HTMLDivElement>
+    specifyThresholds?: boolean;
+    date?: Date;
+}
+
+export interface RelativeActivityDataType {
+    dailyDataType: string;
+    label: string;
+    icon: React.ReactElement;
+    color?: ColorDefinition;
+    overThresholdColor?: ColorDefinition;
+    formatter: (number: number) => string;
+    threshold?: number;
+}
+
+interface RelativeActivityResult {
+    dataType: RelativeActivityDataType;
+    fillPercent: number;
+    value: string;
+}
+
+export default function (props: RelativeActivityProps) {
+    let [result, setResults] = useState<RelativeActivityResult[] | null>(null);
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let date = props.date;
```

Comment:
I think this can simplify to:
`let date = props.date ?? dateRangeContext?.intervalStart ?? startOfDay( new Date() );`